app/lib/actions/pulp3/repository/update_remote.rb

```diff
@@ -0,0 +1,16 @@
+module Actions
+  module Pulp3
+    module Repository
+      class UpdateRemote < Pulp3::Abstract
+        def plan(repository, smart_proxy)
+          plan_self(:repository_id => repository.id, :smart_proxy_id => smart_proxy.id) if repository.root.url?
+        end
+
+        def run
+          repo = ::Katello::Repository.find(input[:repository_id])
+          output[:response] = repo.backend_service(smart_proxy).update_remote
```

Comment:
A remote can't have an empty feed url, so if the feed url is nil, we either need to:
1) delete the remote (and recreate it if it isn't nil)
2) Don't update the remote, and just leave the old url as is.  Katello prevents this from being synced, so it should be okay, but leaves dirty data. 

2) is simpler code wise but not as clean as 1).    I guess i lean towards 1) but curious your thoughts.

Reply:
I would lean away from keep dirty data around.